core/src/main/scala/better/files/ThreadBackedFileMonitor.scala

```diff
@@ -60,15 +57,12 @@ abstract class ThreadBackedFileMonitor(val root: File, maxDepth: Int) extends Fi
     }
   }
 
-  override def start() = {
+  override def start()(implicit executionContext: ExecutionContext) = {
     watch(root, maxDepth)
-    thread.start()
+    executionContext.execute(() => Iterator.continually(service.take()).foreach(process))
```

Comment:
I had been playing with a version where the `ThreadBackedFileMonitor` has a single thread to continually call `take` but the "event handling" bit is scheduled on an executor. A bit like how some HTTP servers will listen to connections then farm off the processing to a pool. I'm not that familiar with the schematics of Scala's `ExecutorContext` so aren't sure how the above operates.